packages/plugin-relay/src/index.ts

```diff
@@ -47,7 +47,7 @@ export class PothosRelayPlugin<Types extends SchemaTypes> extends BasePlugin<Typ
           String(globalID),
           ctx,
           info,
-          Array.isArray(mappings.value) ? mappings.value : false,
+          Array.isArray(mappings.value) ? mappings.value : true,
```

Comment:
I think this will result in always parsing IDs, which would be ideal of we could keep the types in sync, unfortunately I don't think that's possible.  When you declare a globalID are without declaring the type it's for, it could be an ID for any type, and don't know how it's parsed.  If we do parse it, we would likely need to type it as unknown.  Keeping this case with IDs being parsed as strings seems like a better compromise.   I think we need to handle node(s) fields by parsing explicitly before calling the load methods rather than depending on the default arg parsing logic.  We could potentially add an argument to globalID fields to indicate they should always be parsed, or do something like for: '*'.

Reply:
> When you declare a globalID are without declaring the type it’s for, it could be an ID for any type, and don’t know how it’s parsed.

Might be misunderstanding, but it seems like it's determining the typename based on unwrapping the ID, and then based on that typename if it has a parse function it runs it through that?

If you change this back to `false` you'll be able to see the tests that are failing that this was aiming to fix. 

> I think we need to handle node(s) fields by parsing explicitly before calling the load methods rather than depending on the default arg parsing logic. 

That sounds like it makes sense, maybe we can create a test case illustrating what you're referring to here so we ensure that's covered / doesn't regress?

Reply:
I'm traveling for the next few weeks, but I should have a little time tonight to look at this.  I'll add some tests to cover those use cases

Reply:
Thanks! No rush, free to delay until you're done traveling